server/services/zigbee2mqtt/utils/convertValue.js

```diff
@@ -14,11 +14,19 @@ function convertValue(feature, state) {
       result = state === 'ON' ? 1 : 0;
       break;
     }
-    default:
+
+    default: {
       if (typeof state === 'string') {
         throw new Error(`Zigbee2mqqt don't handle value "${state}" for feature "${feature}".`);
       }
-      result = state;
+      // General case : true or false => 1 or 0
+      if (typeof state === 'boolean') {
+        result = state ? 1 : 0;
+      }
+      if (typeof state === 'number') {
+        result = state;
+      }
+    } 
```

Comment:
Maybe it will be clearer to use `switch (typeof state)` ?
```javascript
switch (typeof state) {
  case 'boolean': {
    result = state ? 1 : 0;
    break;
  }
  case 'number': {
    result = state;
    break;
  }
  default:
    throw new Error(`Zigbee2mqqt don't handle value "${state}" for feature "${feature}".`);
}
```

Reply:
I agree